pkg/ctrdtaskapi/update.go

```diff
@@ -15,3 +16,10 @@ type PolicyFragment struct {
 	// fragment and any additional information required for validation.
 	Fragment string `json:"fragment,omitempty"`
 }
+
+type ContainerMount struct {
```

Comment:
Do we plan to put this type in the official container runtime spec in the future?

Reply:
Hmm I am not sure we want to yet - Upstream in the past doesn't seem to have been keen on supporting mounts to running containers. If we have to add it to WindowsResources in oci spec that might be the expectation?  Maybe I can bring up this conversation with upstream folks and see what they have to say and decide if this needs to go under WindowsResources?

Reply:
whats upstream's reasoning for resisting adding mounts to a running container?
do they think the container should be re-created with the new mounts?

Reply:
yes - that is the conversation I have seen before on github issues. But I don't know if this conversation has happened in recent times though. I can bring this up and see if they have any concerns with adding to windows specific resources.
Another thing I also notice in external CRI vs internal CRI is that mountType is not set on upstream CRI like our internal version (not sure if we missed adding this when hyperV support was added or it as intentional - do any of you know why?). A lot of hcsshim code relies on the mountType even in the create container path.

Reply:
Apart from SqlMi, we haven't had other requests for adding mounts to running containers. So not sure how much of an ROI this would be in terms of convincing upstream this is needed and make the necessary changes. I could add this discussion to the next containerd community meeting and hear their thoughts if we are convinced that we want this upstreamed too

Reply:
> Another thing I also notice in external CRI vs internal CRI is that mountType is not set on upstream CRI like our internal version

I think cause process isolated containers just have mapped directories, and upstream assumes mount type is really only for non-windows platforms
https://github.com/microsoft/hcsshim/blob/4dc2c8b9f720ccfce9b86cd0dab5101099ec6849/internal/hcsoci/hcsdoc_wcow.go#L61-L70

Reply:
But how we mount for hyperV wcow is different based on the mount type right? example - https://github.com/microsoft/hcsshim/blob/4dc2c8b9f720ccfce9b86cd0dab5101099ec6849/internal/hcsoci/hcsdoc_wcow.go#L75 ,  https://github.com/microsoft/hcsshim/blob/4dc2c8b9f720ccfce9b86cd0dab5101099ec6849/internal/hcsoci/resources_wcow.go#L148

Reply:
does upstream support hyper-v isolated? i thought they didnt

Reply:
Danny added support for hyperV wcow in containerd/1.7 so its supported

Reply:
> > Another thing I also notice in containerd CRI is that mountType is not set
> 
> I think cause process isolated containers just have mapped directories, and upstream assumes mount type is really only for non-windows platforms
> 
> https://github.com/microsoft/hcsshim/blob/4dc2c8b9f720ccfce9b86cd0dab5101099ec6849/internal/hcsoci/hcsdoc_wcow.go#L61-L70

Checked with Kevin on this and unless there is an ask to support scsi devices from containerd windows community, we need now add this to containerd